src/TeamStore.Keeper/Services/AssetService.cs

```diff
@@ -193,6 +194,39 @@ IApplicationIdentityService applicationIdentityService
             return retrievedAssets;
         }
 
+        // searching assets should preload all assets
+        public async Task<List<Asset>> GetAssetResultsAsync(string searchToken)
+        {
+            // Validate current user
+            var currentUser = await _applicationIdentityService.GetCurrentUser();
+            if (currentUser == null) throw new Exception("Unauthorised requests are not allowed."); ;
+
+            // Get assets with appropriate access checks
+            var retrievedAssets = await _dbContext.Assets.Where(a =>
+                a.IsArchived == false &&
+                a.Project.AccessIdentifiers.Any(ai => ai.Identity.Id == currentUser.Id))
+                .ToListAsync();
+
+            // LOG access asset - open project? TODO
+            var assets = new List<Asset>();
+            foreach (var asset in retrievedAssets)
+            {
+                var title = _encryptionService.DecryptString(asset.Title);
+                if (title.ToLower().Contains(searchToken.ToLower()))
```

Comment:
I now understand that this is probably the best design we can do. 

Do a ToLowerInvariant()